src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,52 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static allAnimals = [];
+  static alive = [];
+
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+    this.hidden = false;
+    Animal.allAnimals.push(this);
+    Animal.alive.push(this);
+  }
+
+  static updateAliveAnimals() {
+    Animal.alive = Animal.allAnimals.filter((animal) => animal.health > 0);
+  }
+
+  checkIfAlive() {
+    if (this.health <= 0) {
+      Animal.alive = Animal.alive.filter((animal) => animal !== this);
+      Animal.allAnimals = Animal.allAnimals.filter((animal) => animal !== this);
+    }
+  }
```

Comment:
The `checkIfAlive` method performs a filter operation on both `Animal.alive` and `Animal.allAnimals` arrays. According to the checklist, you should avoid nested loops and use filter instead. However, this method is called every time an animal's health changes, which could lead to performance issues if there are many animals. Since you already have a filter in place in the `updateAliveAnimals` method, you can remove this method to prevent unnecessary filtering and rely on `updateAliveAnimals` to maintain the `Animal.alive` array.